R/mouse/mouse_cell_type_mapping.R

```diff
@@ -0,0 +1,131 @@
+#' Functions for mapping cell types between deconvolution methods.
+
+require(tidyverse)
+require(magrittr)
+require(rlang)
+require(ComICS)
+
+
+#' Since DCQ and BASE provide estimates for several cell types, this function 
+#'    combines the results to align them with the rest of the methods
+#'
+#' @param deconvolution.results a dataframe containing the results of the 
+#'    deconvolution experiment (DCQ or BASE). Cell types are in columns, samples in rows.
+#' @param annotation which annotation has to be used. Possible values are 'dcq', 'base' or 
+#'    'user' for user provided annotations
+#' @param cell.table a dataframe with two columns: prefixes of cell names and 
+#'    the corresponding cell type. Users can submit their own as well. 
+#' @param method the method used to combine the estimates: sum, for regression-based 
+#'    methods as DCQ, or median for the scoring methods such as BASE.   
+#' @return a data frame with the aggregated cell types     
+#' @export
+
+reduce_mouse_cell_types <- function(deconvolution.results, 
```

Comment:
Any chance you could just re-use the `map_result_to_celltypes` function here? 
https://github.com/icbi-lab/immunedeconv/blob/c70539f2b08901687561dca755337fc6a5130440/R/cell_type_mapping.R#L172-L188
It first translates cell-types to a "controlled vocabulary" and then summarizes cell-types based on the cell-type tree. 

That would imply adding the cell-type mapping to this excel sheet instead of the rds files which would also be great for consistency: 
https://github.com/icbi-lab/immunedeconv/blob/master/inst/extdata/cell_type_mapping.xlsx

Reply:
I don't know if it is manageable without rewriting the function, since here we have to deal with cell types identified by their markers, which are transformed and grouped considering their prefix. I'll make an example: 

T.8sp24..TH -> CD8 T cell

Therefore what I store in the .rds file is just a matrix with the prefixes and the cell type they represent (see the figure). I could eventually save the text file and read them each time, but I thought it was quicker with the .rds since these functions are used wuite often
![immagine](https://user-images.githubusercontent.com/91564922/149542247-765fe45c-c3fc-46e4-b10e-a4aab1ef93d8.png)

In addition to that, the scores can be aggregated either by median or by mean, depending on the method used. 
For all these reasons, I think it would be better to keep the two functions independent